src/runtime_src/core/tools/common/ReportXmc.cpp

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (C) 2021 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "ReportXmc.h"
+#include "core/common/query_requests.h"
+
+void
+ReportXmc::getPropertyTreeInternal( const xrt_core::device * _pDevice,
+                                              boost::property_tree::ptree &_pt) const
+{
+  // Defer to the 20202 format.  If we ever need to update JSON data,
+  // Then update this method to do so.
+  getPropertyTree20202(_pDevice, _pt);
+}
+
+void
+ReportXmc::getPropertyTree20202( const xrt_core::device * _pDevice,
+                                           boost::property_tree::ptree &_pt) const
+{
+  boost::property_tree::ptree pt;
+  try {
+    pt.put("Description", "XMC status");
+    pt.put("heartbeat_err_time", xrt_core::device_query<xrt_core::query::heartbeat_err_time>(_pDevice));
+    pt.put("heartbeat_count", xrt_core::device_query<xrt_core::query::heartbeat_count>(_pDevice));
+    pt.put("heartbeat_err_code", xrt_core::device_query<xrt_core::query::heartbeat_err_code>(_pDevice));
+    pt.put("heartbeat_stall", xrt_core::device_query<xrt_core::query::heartbeat_stall>(_pDevice));
+  } catch(...) {}

```

Comment:
Never do this kind of catch. @stsoe has created a new exception `xrt_core::query::error` which catches errors related to query requests (no_such_key, sysfs_error, a few more), but does not catch unrelated "real" errors.